rotkehlchen/chain/evm/decoding/spark/savings/decoder.py

```diff
@@ -0,0 +1,321 @@
+            base_tools: 'BaseDecoderTools',
+            msg_aggregator: 'MessagesAggregator',
+            spark_savings_tokens: tuple['ChecksumEvmAddress', ...],
+            psm_address: 'ChecksumEvmAddress | None' = None,
+    ) -> None:
+        super().__init__(
+            evm_inquirer=evm_inquirer,
+            base_tools=base_tools,
+            msg_aggregator=msg_aggregator,
+        )
+        self.psm_address = psm_address
+        self.spark_savings_tokens = spark_savings_tokens
+
+    def _decode_psm_deposit_withdrawal(self, context: DecoderContext) -> DecodingOutput:
+        """Decodes PSM (Peg Stability Module) deposit and withdrawal events.
+        PSM allows swapping between assets and their wrapped Spark Savings equivalents.
+        """
+        if context.tx_log.topics[0] != SWAP_TOPIC:
+            return DEFAULT_DECODING_OUTPUT
+
+        asset_in = self.base.get_or_create_evm_token(bytes_to_address(context.tx_log.topics[1]))
+        asset_out = self.base.get_or_create_evm_token(bytes_to_address(context.tx_log.topics[2]))
+        amount_in = token_normalized_value_decimals(
+            token_amount=int.from_bytes(context.tx_log.data[32:64]),
+            token_decimals=asset_in.decimals,
+        )
+        amount_out = token_normalized_value_decimals(
+            token_amount=int.from_bytes(context.tx_log.data[64:96]),
+            token_decimals=asset_out.decimals,
+        )
+        for event in context.decoded_events:
+            if (
+                event.event_type == HistoryEventType.SPEND and
+                event.event_subtype == HistoryEventSubType.NONE and
+                event.amount == amount_in and
+                event.asset == asset_in
+            ):
+                event.counterparty = CPT_SPARK
+                if asset_in.evm_address not in self.spark_savings_tokens:
+                    event.event_type = HistoryEventType.DEPOSIT
+                    event.event_subtype = HistoryEventSubType.DEPOSIT_FOR_WRAPPED
+                    event.notes = f'Deposit {amount_in} {asset_in.symbol} in Spark Savings'
+                else:  # swap exact out
+                    event.event_subtype = HistoryEventSubType.RETURN_WRAPPED
+                    event.notes = f'Return {amount_in} {asset_in.symbol} into Spark Savings'
+
+            elif (
+                event.event_type == HistoryEventType.RECEIVE and
+                event.event_subtype == HistoryEventSubType.NONE and
+                event.amount == amount_out and
+                event.asset == asset_out
+            ):
+                event.counterparty = CPT_SPARK
+                if asset_out.evm_address in self.spark_savings_tokens:
+                    event.event_type = HistoryEventType.RECEIVE
+                    event.event_subtype = HistoryEventSubType.RECEIVE_WRAPPED
+                    event.notes = f'Receive {amount_out} {asset_out.symbol} from depositing into Spark Savings'  # noqa: E501
```

Comment:
if the sUSDS  is decoded via Sky instead of Spark you also show the Spark savings in the comment? Is not clear to me if spark is the oly way to get it now